spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/sql/TestPartitionedTable.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.sql;
+
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.spark.SparkCatalogTestBase;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+
+public class TestPartitionedTable extends SparkCatalogTestBase {
```

Comment:
I think it is good to have this test, I would probably move it to one of the existing suites though, either TestMetadataTablesWithEvolution or TestSelect or just something that already exists since I don't think this needs it's own class.

In addition I think we really should have a test in "core" since that's where we actually hit this, Spark isn't really required for this particular failure so maybe add a test to TestMetadataTableScans? Or a test on PartitionSpecBuilder itself if you change the builder?

Reply:
I moved the test to `TestMetadataTablesWithEvolution` and also add a test in `TestMetadataTableScans`.